### PR TITLE
fozzie-components@v7.30.0 - npm-publish github action

### PR DESCRIPTION
**Pull request by jamieomaguire**
v7.30.0
------------------------------
*September 14, 2022*

### Added
- NPM publish to GitHub Actions.

Screenshot 2022-09-22 at 11 17 11

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -213,3 +214,42 @@ jobs:
...
+            cache: 'yarn'
+        # Restore node_modules - Cache should exist as one was created in previous 'install' job
+        - name: Restore Cache -  Node Modules
+          id: cache-node-modules
+          uses: actions/cache@v2
+          with:
+            path: '**/node_modules'
+            key: node-modules-${{ hashFiles('**/yarn.lock') }}
+        # Run 'yarn' if cache doesn't exist. Use --prefer-offline to download packages from yarn cache folder where possible
+        - name: Install Dependencies
```
Comment (siggerzz) [inline .github/workflows/build.yml]: 237-239 can go, as we're guaranteed to have a node_modules cache 👍🏼

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -213,3 +214,42 @@ jobs:
...
+          id: turborepo-dist-cache
+          uses: actions/cache@v2
+          with:
+            path: .turbo
+            key: turborepo-dist-cache-${{ github.ref }}-${{ github.sha }}
+            # Use one of the following keys if the above is not found.
+            restore-keys: |
+              turborepo-dist-cache-${{ github.ref }}
+              turborepo-dist-cache-
+        - run: npm test
```
Comment (siggerzz) [inline .github/workflows/build.yml]: Might be worth changing this to `yarn build`

I think even though we add the turborepo cache, `yarn build` gets turborepo to move the `dist` folders to the correct place.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -213,3 +214,42 @@ jobs:
         yarn storybook:deploy
       env:
         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  publish:
+      runs-on: ubuntu-latest
```
Comment (siggerzz) [inline .github/workflows/build.yml]: you _should_ be able to do:


`needs: [tests-atoms, tests-molecules, .....]` 

etc to ensure that all test jobs have completed before publishing 👍🏼

File: .github/workflows/build.yml (hunk 2 above)
Comment (jamieomaguire) [reply .github/workflows/build.yml]: sorry - which lines specifically should I change to yarn build?

**Review by siggerzz — DISMISSED** [review]

**Review by siggerzz — DISMISSED** [review]

**Review by kevinrodrigues — DISMISSED** [review]

**Review by siggerzz — DISMISSED** [review]

**Review by kevinrodrigues — DISMISSED** [review]

**Pull request merged**